modules/users/userQueryBulider.js

```diff
@@ -0,0 +1,86 @@
+const DB = require('../../models');
+const bcrypt = require('bcryptjs');
+
+var salt = bcrypt.genSaltSync(10);
+
+const getUser = async (req) => {
+  let query = {
+    limit: req.query.limit || 10,
+    page: req.query.page || 1,
+    sortKey: req.query.sortKey || 'firstName',
+    sortOrder: req.query.sortOrder || 'asc',
+  };
+
+  let userData = await DB.user.findAndCountAll({
+    offset: query.limit * (query.page - 1),
+    limit: query.limit,
+    order: [[query.sortKey, query.sortOrder]],
+  });
+  return {
+    metaData: {
+      page: query.page,
+      perPage: query.limit,
+      totalCount: userData.count,
+      totalPage: Math.ceil(userData.count / query.limit),
+      sortKey: query.sortKey,
+      sortOrder: query.sortOrder,
+    },
+    records: userData.rows,
+  };
+};
+
+const postUser = async (req) => {
+  var hash = bcrypt.hashSync(req.body.password, salt);
+  const role = await DB.role.findOne({ where: { name: req.body.roleId } });
```

Comment:
Get role as role in body or get roleId as roleId in body
This is confusing
Better to get roleId in body